.github/workflows/scripts/cleanup-demo-images.sh

```
@@ -0,0 +1,274 @@
+            fi
+        done
+        shopt -u nullglob
+        
+        # Use git-filter-repo to rewrite history, keeping only desired files
+        if [ -n "$keep_pattern" ]; then
+            log_info "Keeping files matching: $keep_pattern"
+            
+            # Create temporary file with files to keep (one per line)
+            local paths_file
+            paths_file=$(mktemp)
+            trap "rm -f '$paths_file'" EXIT
```

Shellcheck (0.10.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)